azavea/echo-locator
# Fix static assets on staging

## Overview

This allows for the frontend static assets bundled by mastarm to be available on staging as part of the django container image build.

### Notes
- Ideally there should be one fix for both deployed and local environments. I found that after changing the context to include `/taui`, copying assets via the Dockerfile, and taking the volume mounts out of the docker-compose yaml, it worked on deployed environments but no longer locally. I could make larger changes so that the assets could be built/bundled before the django container build in `scripts/update` (so that it better mirrored `scripts/cibuild`), but knowing this will eventually need to be rebased and altered with the new bundler PR #457, I opted to leave the volume/bind mounting in for local dev for now since it will be easier to take out.
- On testing staging, the address field in user profile is not functioning. It looks like the access token we use for mapbox geocoding the address field in user profile is undefined and now we can't submit destinations. This has been addressed in issue #487


## Testing Instructions

 * Delete any already built assets and their directory folder from your local environment: `taui/assets` and `src/backend/static`
 * _optional_ Run `docker system prune`
 * Run` scripts/update`
 * Run `scripts/server` and test static files locally: http://localhost:8085/
 * Navigate to https://stg.echosearch.org/ and confirm frontend displays and you can reach https://stg.echosearch.org/profile
 
Resolves #471, #459
